lib/Parser/Parser.cpp

```diff
@@ -226,15 +251,23 @@ Token Lexer::getNextToken(std::string &ErrStr) {
       ErrStr = "invalid more knownbits string";
       return Token{Token::Error, Begin, 0, APInt()};
     }
+    if (Begin != NumBegin && NumSignBitsFlag && *Begin != ')') {
+      ErrStr = "invalid number of sign bits string";
+      return Token{Token::Error, Begin, 0, APInt()};
+    }
     if (Begin == NumBegin) {
-      ErrStr = "invalid, expected [0|1|x]+ or [n|z|p]";
+      ErrStr = "invalid, expected [0|1|x]+ or [n|z|p] or [s=][0-9]+";
```

Comment:
this isn't right: [s=][0-9]+

this is a regex that would match, for example, '=' or 's' or 's34'